replace SideNav 'closed' class to 'open'

AndreJesusBrito opened the pull request:
## Pull Request Type
<!-- Please select what type of pull request this is: [x] -->
- [X] Bugfix
- [ ] Feature Implementation
- [ ] Documentation
- [x] Other

## Related issue
<!-- Please link the issue your pull request is referring to. -->
<!-- If this pull request fully resolves the relevant issue, put "closes" before the issue number. -->
closes #6091

## Description
This pull request is mostly a refactor and doesn't have any visual or functional effect besides fixing the mentioned issue.

The 'open' state now becomes the default for styling. This has three advantages:
- This makes it consistent with the rest of the application logic (isOpen)
- Simplifies styling. The open styles can now be done for desktop easily with a media query
- Fixes #6091 [Bug]: Theme Settings > Expand side bar by default is affecting the bottom navigation for mobile devices 


## Testing
Ensure that the interface behaves as before when the app resolution (media queries), nav bar open toggle and the Theme > Hide Side Bar Labels option vary.

## Desktop
<!-- Please complete the following information-->
- **OS:** Arch Linux
- **OS Version:** N/A
- **FreeTube version:** v0.23.12-beta

absidue reviewed (commented):
This will conflict with #8365 and as that already has two approvals lets get that merged first, please also make sure this doesn't regress any of the RTL/LTR fixes in that pull request after it is merged:
- LTR channel name + LTR locale -> name is displayed LTR
- LTR channel name + RTL locale -> name is displayed LTR
- RTL channel name + LTR locale -> name is displayed RTL
- RTL channel name + RTL locale -> name is displayed RTL

AndreJesusBrito committed:
replace SideNav 'closed' class to 'open'

The 'open' state now becomes the default for styling. This has three
advantages:
- This makes it consistent with the rest of the application logic (isOpen)
- Simplifies styling. The open styles can now be done for desktop
  easily with a media query
- Fixes #6091 [Bug]: Theme Settings > Expand side bar by default is affecting the bottom navigation for mobile devices

AndreJesusBrito committed:
rename 'open' css class to 'opened' in SideNav

AndreJesusBrito committed:
Sidenav: removed navChannel right margin when opened

PikachuEXE reviewed (commented):
I tested briefly but I got no idea what I should watch out for

efb4f5ff-1298-471a-8973-3d47447115dc commented:
@PikachuEXE did you see linked issue https://github.com/FreeTubeApp/FreeTube/issues/6091?

PikachuEXE commented:
#6091 is tested I mean what else :)

AndreJesusBrito committed:
refactor(SideNav): removed redundant margin css rules

AndreJesusBrito commented:
Hi @efb4f5ff-1298-471a-8973-3d47447115dc and @PikachuEXE , I noticed that the PR was automatically closed. Let me know if you still interested in merging this.

Regarding the margins, I found that some of those were redundant. 
I made a new commit removing them, but I think it doesn't show here because the PR is closed.
Also the margin rules were already present in the file before my changes, I only moved the lines around to "invert" from the __closed__ class to an __open__ class.

efb4f5ff-1298-471a-8973-3d47447115dc commented:
@AndreJesusBrito ill reopen the PR. Please double check in our previous conversations to see if you implemented all requested changes

AndreJesusBrito committed:
fix: Channels icon too big

AndreJesusBrito commented:
@efb4f5ff-1298-471a-8973-3d47447115dc I doubled checked all and I noticed the channels icon was bigger than before (more 2px). Made a fix and now everything looks good

absidue reviewed (approved):
Thank you for your pull request!! We are sorry it took so long to get it merged!

AndreJesusBrito commented:
No worries! Looking to make some PRs in the future when a little more free :)
